ADALiOS/ADALiOS/ADTokenCacheItem.m

```diff
@@ -17,16 +17,33 @@
 // governing permissions and limitations under the License.
 
 #import "ADALiOS.h"
-#import "ADTokenCacheStoreItem.h"
+#import "ADTokenCacheItem.h"
 #import "ADUserInformation.h"
 #import "ADOAuth2Constants.h"
 #import "ADAuthenticationSettings.h"
-#import "ADTokenCacheStoreKey.h"
+#import "ADTokenCacheKey.h"
 
-@implementation ADTokenCacheStoreItem
+@implementation ADTokenCacheItem
+{
+    NSUInteger _hash;
+    NSString* _resource;
+    NSString* _authority;
+    NSString* _clientId;
+    ADUserInformation* _userInformation;
+}
 
 @synthesize multiResourceRefreshToken;
 
+- (NSUInteger)hash
+{
+    return _hash;
+}
+
+- (void)calculateHash
+{
+    _hash = [[NSString stringWithFormat:@"%@%@%@%@", _resource, _authority, _clientId, _userInformation.userId] hash];
```

Comment:
It doesn't matter. This doesn't need to duplicate ADTokenCacheKey (if you notice, there are more fields here...) this is trying to pull out an integer value that can help sort the object, by definition it's not going to have the same resolution as the string that was hashed.